cmake/Endian.cmake

```diff
@@ -3,12 +3,23 @@
 # add_endian_definition()
 #
 function(add_endian_definition)
-    include(TestBigEndian)
-    TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
+    
+    if(CMAKE_VERSION VERSION_LESS "3.20.0") 
+        include(TestBigEndian)
+        TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
+        if(IS_BIG_ENDIAN)
+            add_definitions(-DTTD_ENDIAN=TTD_BIG_ENDIAN)
+        else()
+            add_definitions(-DTTD_ENDIAN=TTD_LITTLE_ENDIAN)
+        endif()
 
-    if(IS_BIG_ENDIAN)
-        add_definitions(-DTTD_ENDIAN=TTD_BIG_ENDIAN)
     else()
-        add_definitions(-DTTD_ENDIAN=TTD_LITTLE_ENDIAN)
+        if(CMAKE_<LANG>_BYTE_ORDER EQUAL BIG_ENDIAN)
```

Comment:
yeah I don't think that's going to work

Reply:
I promise it doesn't. `<LANG>` is a placeholder in the documentation.

Reply:
`CMAKE_<LANG>_BYTE_ORDER` is a generic placeholder for documentation, you want to use `CMAKE_CXX_BYTE_ORDER` (and `CMAKE_OBJCXX_BYTE_ORDER` for .mm files)

Reply:
> Works for me.

It 'works' for you because you do not have a BIG_ENDIAN machine, so it doesn't matter if the condition is always false.